### ISSUE TITLE
Add to list and events not linking

### ISSUE THREAD

**Issue by andrewcgregory** [FosterCommerce/klaviyoconnect#9]
Having an issue when trying to add a user to a Klaviyo List as well as tracking Events.

The website flow is that the visitor goes to the cart and then goes to a sign-in/register page. On the register form they enter their email address and then submit the form (this is the form that we are trying to use with Klaviyo Connect). The visitor is then sent to a shipping/billing information page.

I tried to follow the template examples at https://klaviyoconnect.fostercommerce.com/Templating.html and currently am using the code below.

As soon as I submit the form I am seeing the new user in the *Profiles* section of Klaviyo, so the connection itself is working. It also includes include the "Updated Cart" event. However, the user is not added to the List and I have not been able to get any other Events to fire either on this or on other forms.

Tested on:
Craft: 3.1.15 (and 3.1.19)
Commerce 2.1.0.2 (and 2.1.2)

```html
<form method="POST">
<input type="hidden" name="action" value="klaviyoconnect/api/track">
<input type="hidden" name="list" value="{{ klaviyo.klaviyoListId }}">
<input type="hidden" name="confirmOptIn" value="0" />
<input type="hidden" name="event[name]" value="Event Foo" />

<input type="hidden" name="action" value="commerce/cart/update-cart">
{{ redirectInput('/checkout/order-details') }}
{{ csrfInput() }}
```

**Comment by johnnynotsolucky** [comment]
@andrewcgregory The templates section hasn't been updated yet since the last batch of updates I've made 🙂

The new user in Profiles and the Updated Cart are triggered automatically when the cart is updated. I think the form you're submitting can be updated:

```twig
<input type="hidden" name="action" value="commerce/cart/update-cart">
```

Should be

```twig
<input type="hidden" name="forward" value="commerce/cart/update-cart">
```

`forward` tells the plugin to forward requests onto the next controller: https://klaviyoconnect.fostercommerce.com/Actions.html#extra-form-parameters

Please let me know if this resolves your question.

**Comment by andrewcgregory** [comment]
@johnnynotsolucky I tried updating the line as you mentioned, so I have:
```html
<input type="hidden" name="action" value="klaviyoconnect/api/track">
<input type="hidden" name="list" value="{{ klaviyo.klaviyoListId }}">
<input type="hidden" name="confirmOptIn" value="0" />
<input type="hidden" name="event[name]" value="Event Foo" />

<input type="hidden" name="forward" value="commerce/cart/update-cart">
{{ redirectInput('/checkout/order-details') }}
{{ csrfInput() }}
```
Looks like this throws a 404 error and no longer follows the redirect? Also, nothing updated to Klaviyo. Here's the error:

```HTTP 404 – Not Found – yii\web\NotFoundHttpException
Page not found.
↵
Caused by: Invalid Route – yii\base\InvalidRouteException
Unable to resolve the request "klaviyoConnect/api/track".
in .../vendor/yiisoft/yii2/base/Module.php at line 537
```

It also shows:
``` 
'redirect' => 'store/checkout/shipping',
```
This path does not match our website, any ideas on where this would be coming from?

**Comment by johnnynotsolucky** [comment]
@andrewcgregory It occurred to me that you're probably using v2.0.6 while the documentation was for 3.0.0 which had not yet been released. That is probably why you're getting the not found error.

I've published v3.0.0 just now and updated the template documentation too.

One thing I missed in my earlier comment was that the action addresses should be absolute, so your code would be something like

```twig
<input type="hidden" name="action" value="/klaviyoconnect/api/track">
<input type="hidden" name="list" value="{{ klaviyo.klaviyoListId }}">
<input type="hidden" name="confirmOptIn" value="0" />
<input type="hidden" name="event[name]" value="Event Foo" />

<input type="hidden" name="forward" value="/commerce/cart/update-cart">
{{ redirectInput('/checkout/order-details') }}
{{ csrfInput() }}
```

RE the weird redirect parameter, I'm not sure where that is coming from, it looks like the standard route created by Commerce when it is first installed.

**Comment by andrewcgregory** [comment]
@johnnynotsolucky That's working better. I did a fresh install to v3.0.0 and used your recommended changes. The user is now added to the list and events are passed as they should be.

One new item that I came across is when accessing **Settings > Plugins > Klaviyo Connect** it is now giving an error and no **Default Profile Mapping** options show up. 

```
Twig Runtime Error – Twig\Error\RuntimeError
Neither the property "profileMappings" nor one of the methods "profileMappings()", "getprofileMappings()"/"isprofileMappings()"/"hasprofileMappings()" or "__call()" exist and have public access in class "fostercommerce\klaviyoconnect\variables\Variable".
```

**Screenshot:**
![Screen Shot 2019-03-28 at 8 39 56 AM](https://user-images.githubusercontent.com/1779776/55159641-ef383480-5137-11e9-8ad0-c35bf990bc51.png)

**Comment by johnnynotsolucky** [comment]
@andrewcgregory this is resolved in v3.0.1
